### ISSUE TITLE
Disconnecting a pinned participant does not update main participant

### ISSUE THREAD

**Issue by dannnnthemannnn** [twilio/twilio-video-app-react#338]
**Describe the bug**
If a user is pinned and then disconnects, it does not update the "main participant" to remove that user. So the screen is black and says their name even though they are not connected. 

**To Reproduce**
Pin someone.
Disconnect that user. (I am doing this via REST api but I'm guessing it happens even if they just close out.)

**Expected behavior**
The main partipant display goes back to dominat speaker or someone else

**Screenshots**

**Environment (please complete the following information):**

**Additional context**
I fixed this by updating useMainParticipant with the below:

```
  const filteredSelectedParticipant = participants.some((p) => p.identity == selectedParticipant?.identity) ? selectedParticipant : null;

  // The participant that is returned is displayed in the main video area. Changing the order of the following
  // variables will change the how the main speaker is determined.
  return filteredSelectedParticipant || remoteScreenShareParticipant || dominantSpeaker || participants[0] || localParticipant;
```

Not sure if the same issue applies to dominantspeaker or remoteScreenShareParticipant

**Comment by timmydoza** [comment]
Thanks for the bug report @dannnnthemannnn!

I have a PR up to fix it: #339 

I updated the `useSelectedParticipant` hook so it unpins participants when they disconnect from a room. This is done by listening to the `participantDisconnected` event.

**Comment by dannnnthemannnn** [comment]
Great! Thanks for the fix!
